hack/README.md

```diff
@@ -1,28 +1,104 @@
 
 ### Usage
+
 `./deploy.sh [username@]<target-server-ip> <consul-ip> [deploy-agent*|deploy-web]`
 
-  - `[username]@<target-server-ip>`
-    The IP address of the host where we are deploying `trento` and `consul` on.
-    
-  - `<consul-ip>`
-    The IP of the consul server that we are connecting to. When `deploy-web` is
-    used in the next field, this is ignored.
-
-  - `[deploy-agent|deploy-web]`
-    `deploy-agent` causes to deploy the `consul` and `trento` agents 
-    while
-    `deploy-web` causes to deploy the web server as well as a `consul` server instance
+- `[username]@<target-server-ip>`
+  The IP address of the host where we are deploying `trento` and `consul` on.
+
+- `<consul-ip>`
+  The IP of the consul server that we are connecting to. When `deploy-web` is used in the next field, this is ignored.
+
+- `[deploy-agent|deploy-web]`
+  `deploy-agent` causes to deploy the `consul` and `trento` agents while
+  `deploy-web` causes to deploy the web server as well as a `consul` server instance
+
+## Automatic deploy to private nodes with GitHub actions
+
+The repository contains a GitHub actions [workflow](../.github/workflows/ci.yaml) to deploy Trento to private nodes by
+using [self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners).
+
+This is triggered automatically when pushing (or merging PRs) into the `main` branch of the upstream remote, or manually
+by
+the [workflow_dispatch](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_dispatch)
+event
+(e.g., when working on forks).
+
+### Set up runners instances
+
+#### Generate token
+
+Refer
+to [this guide](https://docs.github.com/en/actions/hosting-your-own-runners/adding-self-hosted-runners#adding-a-self-hosted-runner-to-a-repository)
+to add a runner to the repository. The generated token will be passed as a parameter to the provisioning script,
+see [Provision](#provision).
+
+Programmatically registering self-hosted runners is also possible by using the rest API.
+See [https://docs.github.com/en/rest/reference/actions#self-hosted-runners](https://docs.github.com/en/rest/reference/actions#self-hosted-runners)
+
+#### Runner labels
+
+The provided actions workflow expects 3 nodes in total: 2 hana nodes and a monitoring node. For the actions to run, the
+runners must have a label specifing the node type (`vmhana01`,`vmhana02` and `vmmonitoring`)
+
+#### Provision
+
+[`provision.sh`](./provision.sh) provides an easy way to boostrap a GitHub self-hosted runner environment on a private
+node.
+
+The script sets up users, configuration files, systemd unit and a self-hosted runner instance required in order to run
+Trento, Consul and the deployment process.
+
+`# ./setup.sh <github-user> <github-repo> <actions-runner-token> <actions-runner-name> <consul-bind-ip> <consul-server-ip> [agent|web]`
```

Comment:
You probably missed to update the name of the script here. This would be `provision.sh` right?